src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -277,5 +378,45 @@ private static string ToHtmlCase(string name)
         {
             return HtmlCaseRegex.Replace(name, HtmlCaseRegexReplacement).ToLowerInvariant();
         }
+
+        private static Type ExtractGenericInterface(Type queryType, Type interfaceType)
```

Comment:
Cribbed this method from MVC repo's Common project.  Will move to Common repo as I make this PR non-design-y, unless I hear objections.